apps/nextjs/src/components/AppComponents/Chat/chat-lessonPlanDisplay.tsx

```diff
@@ -62,54 +72,113 @@ const organiseSections = [
 ];
 
+function getSectionsToDisplay(
+  lessonPlanKeys: LessonPlanKeys[],
+  streamingSections: LessonPlanKeys[] | undefined,
+) {
+  const sectionsFromOrganiseSections = organiseSections.flatMap((section) => {
+    const trigger = section.trigger;
+    if (
+      lessonPlanKeys.includes(trigger) ||
+      streamingSections?.includes(trigger)
+    ) {
+      return section.dependants.filter(
+        (dependant) =>
+          streamingSections?.includes(dependant) ||
+          lessonPlanKeys.includes(dependant as LessonPlanKeys),
+      );
+    }
+    return [];
+  });
+
+  const sectionsFromAllSectionsInOrder = allSectionsInOrder.filter((section) =>
+    lessonPlanKeys.includes(section as LessonPlanKeys),
+  );
+
+  const sectionsToDisplay = [
+    ...new Set([
+      ...sectionsFromOrganiseSections,
+      ...sectionsFromAllSectionsInOrder,
+    ]),
+  ];
+  return sectionsToDisplay;
+}
+
 export const LessonPlanDisplay = ({
   chatEndRef,
-  sectionRefs,
-  documentContainerRef,
   showLessonMobile,
 }: {
   chatEndRef: React.MutableRefObject<HTMLDivElement | null>;
-  sectionRefs: Record<string, React.MutableRefObject<HTMLDivElement | null>>;
-  documentContainerRef: React.MutableRefObject<HTMLDivElement | null>;
   showLessonMobile: boolean;
 }) => {
-  const chat = useLessonChat();
-  const { lessonPlan, ailaStreamingStatus, lastModeration } = chat;
+  const [openSections, setOpenSections] = useState<Record<string, boolean>>({});
+  const [sectionsToDisplay, setSectionsToDisplay] = useState<LessonPlanKeys[]>(
+    [],
+  );
 
-  const [userHasCancelledAutoScroll, setUserHasCancelledAutoScroll] =
-    useState(false);
+  const handleSetIsOpen = (section: string, isOpen: boolean) => {
+    setOpenSections((prevState) => ({
+      ...prevState,
+      [section]: isOpen,
+    }));
+  };
+
+  const chat = useLessonChat();
+  const {
+    lessonPlan,
+    ailaStreamingStatus,
+    lastModeration,
+    streamingSection,
+    streamingSections,
+    setSectionRef,
+  } = chat;
+
+  const lessonPlanSectionKeys = useMemo(
+    () =>
+      Object.keys(lessonPlan).filter((key) =>
+        allSectionsInOrder.includes(key as LessonPlanKeys),
+      ) as LessonPlanKeys[],
+    [lessonPlan],
+  );
 
+  // If a section is temporarily missing, we don't suddenly
+  // hide the section until it appears again
   useEffect(() => {
-    const handleUserScroll = () => {
-      // Check for mousewheel or touch pad scroll event
-      event?.type === "wheel" && setUserHasCancelledAutoScroll(true);
-    };
-
-    if (ailaStreamingStatus === "Idle") {
-      // hack to account for lag
-      const timer = setTimeout(() => {
-        setUserHasCancelledAutoScroll(false);
-      }, 1000);
-      return () => clearTimeout(timer);
-    }
+    const newSectionsToDisplay = getSectionsToDisplay(
+      lessonPlanSectionKeys,
+      streamingSections,
+    );
+    setSectionsToDisplay((prevSectionsToDisplay) => {
```

Comment:
Reduces some thrash about which sections to display. But if we were to just use the keys that are on the lesson or are streaming in, we could remove all this.